main.go

```diff
@@ -412,7 +413,11 @@ func main() {
 	if cfg.Once {
 		err := ctrl.RunOnce(ctx)
 		if err != nil {
-			log.Fatal(err)
+			if strings.Contains(err.Error(), "Throttling: Rate exceeded") {
```

Comment:
Let's try to use the AWS SDK's proposed error handling describe here: https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/handling-errors.html

```go
if err != nil {
    if awsErr, ok := err.(awserr.Error); ok {
        // process SDK error
    }
}
```

There should already be similar code examples in this project for inspiration.

Reply:
Since we are handling a _specific_ AWS error using the code above (even when we use string matching since it's fairly specific) this code should not be in the part of code that's shared between all providers (such as `main.go`, `controller.go` etc.).

In needs to be somewhere in the `provider/aws` package. So I think if we want to do it this way we need to handle the rate limits in the AWS providers functions and _not_ return an error in this case (merely log instead) such that the [control loop](https://github.com/kubernetes-sigs/external-dns/blob/master/controller/controller.go#L314) won't exit the program.

Reply:
I just noticed that _not_ returning an error isn't really an option because that would mean you have to return some result, such as the lit of records, which you don't have.

Instead in the [control loop](https://github.com/kubernetes-sigs/external-dns/blob/master/controller/controller.go#L314) we can merely log the error instead of fataling, so that the program isn't terminated and waits for next sync period.

This would change existing behaviour that people could rely on (e.g. use process termination and exit code in their monitoring). But we could make it configurable, e.g. `--fail-on-error=[true|false]`.